src/java.base/share/classes/java/text/SimpleDateFormat.java

```diff
@@ -1425,7 +1444,11 @@ private void zeroPaddingNumber(int value, int minDigits, int maxDigits, StringBu
 
         numberFormat.setMinimumIntegerDigits(minDigits);
         numberFormat.setMaximumIntegerDigits(maxDigits);
-        numberFormat.format((long)value, buffer, DontCareFieldPosition.INSTANCE);
+        if (buffer.isProxyStringBuilder()) {
+            numberFormat.format((long)value, buffer.asStringBuilder(), DontCareFieldPosition.INSTANCE);
```

Comment:
This `numberFormat` might be a user class; if that's the case, we might do something like:
```java
if (buffer.isProxyStringBuilder()) {
    var nf = numberFormat; // field is protected, users can change it even with races!
    if (nf.isInternalSubclass()) {
        nf.format((long)value, buffer.asStringBuilder(), DontCareFieldPosition.INSTANCE);
    } else {
        // format to stringbuffer, and add that buffer to the builder
        buffer.append(nf.format((long)value, new StringBuffer(), DontCareFieldPosition.INSTANCE));
    }
} else { // existing code
```